packages/common/src/api.ts

```diff
@@ -17,4 +19,37 @@ export function getHubApiUrl(requestOptions: IRequestOptions): string {
   
 }
 
+export function getFromHubAPI(id:string, hubRequestOptions?: IHubRequestOptions): Promise<any> {
```

Comment:
We'll have to figure out how to handle an optional HubRequestOptions - b/c that is how we know if we're dealing with portal or not. And in fact, determining that for an un-authenticated is a real problem, which we get around by having different builds of UI which have that flag "built-in" as part of the app's config. For the "Content" aspect (and likely other aspects of Hub.js) we need some solution, which may end up being Classes composed from lower-level functions... and the constructor requires either a full `UserSession` or a more minimal hash with the portal url and a `isPortal` flag... something like that

Reply:
That's doable, as long as we compose the `API` out of the free functions. Another wrinkle  is that the UI application uses two identities in a number of places, so we want to make it easy/cheap to spin up an instance (i.e. we don't fire off N xhrs in the factory)

Reply:
First, determining the environment would seem to remain as a deployment configuration / build config. When a deploy is finished it can copy/write the config.json file with the environment parameters that is fetched at app bootup. 

@mikeringrose within that, I think your proposal is then to wrap that state management/access into an `API` module. Is that accurate?

There might be various cases where there are different environment parameters such as calling to another Org (community org) or even to an Enterprise server. These would be handled within that specific workflow passing in unique options. For the community use-case the "Portal URL" and user credentials are fetched at operation time such as "search for all (private) community members"

Reply:
@mikeringrose's suggestion is intriguing and worth investigating. Assuming the `hubAPI` instance would expose all fns w/ _overridable_ defaults, it's probably a less hacky version of what [we did in rest-js](https://esri.github.io/arcgis-rest-js/api/request/setDefaultRequestOptions/), which we don't yet take advantage of in the Hub app. Seems like it could be a lot of work/maintenance to attach all the fns to the instance tho. I'm curious to see what it looks like in practice.

Reply:
Thinking about this some more, @mikeringrose's suggestion (as I understand it) would likely prevent us from having tree shaking or lazy-loading certain portions of the app.

Full disclosure, it's very possible that the hack I use for rest-js' `setDefaultRequestOptions()` technically violates [webpack's `sideEffects: false` convention](https://webpack.js.org/guides/tree-shaking/#mark-the-file-as-side-effect-free), but we've tested and it seems to work, so 🤷‍♂️